### ISSUE TITLE
btrfs subvolume list shows a "TOPLEVEL" subvolume after removing it

### ISSUE THREAD

**Issue by vojtechtrefny** [kdave/btrfs-progs#72]
After removing a subvolume, running "btrfs subvolume list" shows a new entry "ID 257 gen 12 top level 0 path TOPLEVEL".

```
# btrfs subvolume list /mnt/test

# btrfs subvolume create /mnt/test/a
Create subvolume '/mnt/test/a'

# btrfs subvolume list /mnt/test
ID 257 gen 12 top level 5 path a

# btrfs subvolume delete -c /mnt/test/a
Delete subvolume (commit): '/mnt/test/a'

# btrfs subvolume list /mnt/test
ID 257 gen 12 top level 0 path TOPLEVEL
```

This entry disappears after ~10 seconds. 

```
# btrfs --version
btrfs-progs v4.13.2

# uname -r
4.13.0-0.rc5.git4.1.fc28.x86_64
```

This doesn't happen with v4.13.1.

We are using _subvolume list_ in our tests to check if our code for removing subvolumes works and this "new subvolume" causes our tests to crash.

**Comment by littleroad** [comment]
Thanks，I have sent a patch to fix it.

https://patchwork.kernel.org/patch/10003917/

**Comment by kdave** [comment]
Thanks.

**Comment by vojtechtrefny** [comment]
Thank you!

**Comment by kdave** [comment]
Released in 4.13.3.
